### ISSUE TITLE
device-width and device-height are deprecated

### ISSUE THREAD

**Issue by ghost** [siddarthvader/CSS-MEDIA-ALL#1]
https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries

> Deprecated
> This feature has been removed from the Web standards. Though some browsers may still support it, it is in the process of being dropped. Avoid using it and update existing code if possible; see the compatibility table at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

**Comment by siddarthvader** [comment]
yes thanks for pointing it out, I will take out time and update the library soon, thanks.

**Comment by ghost** [comment]
Here comes. How are you going to update the library without those features, reliably being able to separate mobile devices from each other?

Have a look at [this question](http://stackoverflow.com/questions/43635632/change-layout-on-orientation-change-navigation-and-li-elements-fixed-or-static) to see what I mean with reliably being able to tell if a device has a certain height or width when held in different orientations.

The way it stands today and from the little I know this cannot be done with pure CSS. Sure landscape and portrait seem to do the trick but how on earth are you going to be able to [separate in iPad from a Glaaxy tab](http://pioul.fr/cross-device-cross-browser-portrait-landscape-detection) with just CSS and possibly with vendors not correctly assigning portrait and landscape orientation?

I am visiting a device lab soon and will make a small site where I can see if purely from CSS portrait and landscape orientation is correctly given like [this reply](http://stackoverflow.com/a/43636252/1010918) seems to indicate.

So my reply is not so much about your library but possibly about how we could together find a way to deal with the plenty of mobile devices and come up with a cross browser and cross device way of being able to separate certain devices from each other, possibly only using CSS.
Is this possible at all?

Sure responsive design is **not** about targeting specific devices with media queries in CSS but to have an overall fluid layout that adjust from 50pxpx upwards to quad HD TVs (sorry, [even 8K UDH TVs even, people are nuts!](https://en.wikipedia.org/wiki/Graphics_display_resolution)) these days.

So yeah, happy to have some ideas going back and forth and seeing what comes of it and for sure, if you come up with a way to update your library I would be very interested to learn about the approach you are taking.

**Comment by siddarthvader** [comment]
i am currently hooked up somewhere, do you want to make the changes in library, seems like you have dig the topic real well? you can put up a pull request
